_articles/abilities/lua-modifiers/5.md

```diff
@@ -0,0 +1,208 @@
+---
+title: Generic Barrier Base Class
+author: Cykada
+steamId: '76561198066443130'
+date: 08.04.2024
+---
+
+## Rundown
+( [just give me the code](https://pastebin.com/M1E2zjku) )
+
+Barriers are a classic case of something dota has had introduced without being fully exposed to the modding API, so I've taken it upon myself to create a generic modifier class that can be inherited from in order to streamline barrier creation, and add some extra functionality to barriers in general. Unfortunately, this won't add any functionality to any native barriers (Aphotic Shield, Shield Rune, Pipe of Insight, the cheese upgrade no-one cared about a week after the update, etc) as since this is for custom barriers.
+
+*Also uses the [Timers](https://github.com/ModDota/Dota2AIFramework/blob/master/Game/scripts/vscripts/Libraries/Timers.lua) library to avoid using the modifier's `OnIntervalThink()` function.*
```

Comment:
```suggestion
*Also uses the [Timers](https://github.com/ModDota/TypeScriptAddonTemplate/blob/master/src/vscripts/lib/timers.lua) library to avoid using the modifier's `OnIntervalThink()` function.*
```

Appreciate the AI framework shoutout but that version is very outdated